cli/cfncluster/cfncluster.py

```diff
@@ -125,58 +131,62 @@ def update(args):
-                                                    aws_secret_access_key=config.aws_secret_access_key)
-    asgconn = boto.ec2.autoscale.connect_to_region(config.region,aws_access_key_id=config.aws_access_key_id,
-                                                    aws_secret_access_key=config.aws_secret_access_key)
+
+    cfn = boto3.client('cloudformation', region_name=config.region,
+                       aws_access_key_id=config.aws_access_key_id,
+                       aws_secret_access_key=config.aws_secret_access_key)
+
+    asg = boto3.client('autoscaling', region_name=config.region,
+                       aws_access_key_id=config.aws_access_key_id,
+                       aws_secret_access_key=config.aws_secret_access_key)
+
     if not args.reset_desired:
-        temp_resources = []
-        resources = cfnconn.describe_stack_resources(stack_name)
-        while True:
-            temp_resources.extend(resources)
-            if not resources.next_token:
-                break
-            resources = cfnconn.describe_stack_resources(stack, next_token=resources.next_token)
-        resources = temp_resources
-        asg = [r for r in resources if r.logical_resource_id == 'ComputeFleet'][0].physical_resource_id
-
-        desired_capacity = asgconn.get_all_groups(names=[asg])[0].desired_capacity
+        resources = cfn.describe_stack_resources(StackName=stack_name).get('StackResources')
+        asg_name = [r for r in resources if r.get('LogicalResourceId') == 'ComputeFleet'][0].get('PhysicalResourceId')
+
+        desired_capacity = asg.describe_auto_scaling_groups(AutoScalingGroupNames=[asg_name])\
+            .get('AutoScalingGroups')[0]\
+            .get('DesiredCapacity')
         config.parameters.append(('InitialQueueSize', desired_capacity))
 
     # Get the MasterSubnetId and use it to determine AvailabilityZone
     try:
         i = [p[0] for p in config.parameters].index('MasterSubnetId')
         master_subnet_id = config.parameters[i][1]
         try:
-            vpcconn = boto.vpc.connect_to_region(config.region,aws_access_key_id=config.aws_access_key_id,
-                                                 aws_secret_access_key=config.aws_secret_access_key)
-            availability_zone = str(vpcconn.get_all_subnets(subnet_ids=master_subnet_id)[0].availability_zone)
-        except boto.exception.BotoServerError as e:
-            logger.critical(e.message)
+            ec2 = boto3.client('ec2', region_name=config.region,
+                               aws_access_key_id=config.aws_access_key_id,
+                               aws_secret_access_key=config.aws_secret_access_key)
+            availability_zone = ec2.describe_subnets(SubnetIds=[master_subnet_id]) \
+                .get('Subnets')[0] \
+                .get('AvailabilityZone')
+        except ClientError as e:
+            logger.critical(e)
```

Comment:
why not having `logger.critical(e.response.get('Error').get('Message'))`

Reply:
changed.